main.py

```diff
@@ -36,23 +38,24 @@ def rest_save(self, namespace, value, timestamp=None):
         try:
             self.backend.save(namespace, value, timestamp)
         except (InvalidNamespaceError, ValueConvertError) as exc:
-            return jsonify({"response": str(exc)})
+            exc_name = exc.__class__.__name__
+            return jsonify({'response': str(exc), 'exc_name': exc_name})
 
-        return jsonify({"response": "Value saved !"})
+        return jsonify({'response': 'Value saved !'})
 
     @rest('v1/<namespace>/', methods=['DELETE'])
     @rest('v1/<namespace>/start/<start>', methods=['DELETE'])
     @rest('v1/<namespace>/end/<end>', methods=['DELETE'])
     @rest('v1/<namespace>/<start>/<end>', methods=['DELETE'])
     def rest_delete(self, namespace, start=None, end=None):
         """Delete the data in one of the backends."""
-        log.info(start)
-        log.info(end)
-        result = self.backend.delete(namespace, start, end)
-        if result in (400, 404):
-            return jsonify({"response": "Not Found"}), 404
+        try:
+            self.backend.delete(namespace, start, end)
+        except (InvalidNamespaceError, ValueConvertError, TimestampRangeError,
+                NamespaceNotExistsError) as exc:
```

Comment:
This reinforces my comment about too-specific exceptions: there's no point in using 4 different exceptions just to check for them all later.

Reply:
Suggestion to reduce the number of custom exception classes:
- ValueConvertError, TimestampRangeError -> ValueError
- InvalidNamespaceError, NamespaceNotExistsError -> InvalidNamespaceError

Another idea would be to Inherit from something like "KronosBaseException".

Reply:
In Kronos the task of convert a value it can face the following error cases: 
- convert an invalid timestamp string (e.g. '1002a') to float
- convert a timestamp value out of supported range (e.g. too big int)
- convert an invalid value field string (e.g. '122a') to float

In that cases kronos main will receive a ValueError or OverflowError, 
exceptions that does not help to understand what problem occurred. It means for
 instance, if any backend or any method in 'utils.py'raise a exception 
ValueError the 'main.py' of Kronos NApp won't know that corresponds to a 
conversion error case. The  ValueConvertError will allow Kronos NApp to 
identify all that cases as a specific case e set more specific error messages.

Reply:
The exceptions InvalidNamespaceError, NamespaceNotExistsError were replaced by NamespaceError.